functions.R

```diff
@@ -0,0 +1,319 @@
+    renamed_variable_to_tabulate_by_one<-names(which(rename_lookup[[database_name]]==variable_to_tabulate_by_one))
+    
+    #allow the variable to be used as a column name
+    renamed_variable_to_tabulate_by_one<- sym(renamed_variable_to_tabulate_by_one) 
+  
+    summary_table <- df_to_create_summary_table |>
+      #separate rows if there are multi select options present
+      separate_rows(!!renamed_variable_to_tabulate_by_one,sep = " {~@~} ") |>
+      #use count to tabulate
+      count(!!renamed_variable_to_tabulate_by_one,.drop= FALSE)|>
+      #  add row totals and percentage column
+      mutate(percent = scales::percent(n / sum(n))) %>%
+      adorn_totals('row')
+    
+    
+  } else if ( length(variables_to_tabulate_by_list)==2){
+    
+    # extract the variables from list of variables
+    variable_to_tabulate_by_one<- unlist(variables_to_tabulate_by_list)[[1]]
+    variable_to_tabulate_by_two<- unlist(variables_to_tabulate_by_list)[[2]]
+    
+    #convert the variables which data will be tabulated to a more human readable name, using lookup
+    renamed_variable_to_tabulate_by_one<-names(which(rename_lookup[[database_name]]==variable_to_tabulate_by_one))
+    renamed_variable_to_tabulate_by_two<-names(which(rename_lookup[[database_name]]==variable_to_tabulate_by_two))
+    
+    #allow the variable to be used as a column name
+    renamed_variable_to_tabulate_by_one<- sym(renamed_variable_to_tabulate_by_one) 
+    renamed_variable_to_tabulate_by_two<- sym(renamed_variable_to_tabulate_by_two) 
+    
+    
+    summary_table <- df_to_create_summary_table |>
+      #separate rows if there are multi select options present
+      separate_rows(!!renamed_variable_to_tabulate_by_one, sep = " {~@~} ") |>
+      separate_rows(!!renamed_variable_to_tabulate_by_two, sep = " {~@~} ") |>
+      # use count to get a table
+      count(!!renamed_variable_to_tabulate_by_one,!!renamed_variable_to_tabulate_by_two,.drop= FALSE)%>% 
+      #pivot so variable 2 is columns
+      pivot_wider(names_from = !!renamed_variable_to_tabulate_by_two,
+                  values_from = n) %>%
+      #add row and column totals
+      adorn_totals('both')
+    
+  } else{
+    message("TOO MANY VARIABLES INCLUDED- ONLY THE FIRST TWO WILL BE USED")
+  }
+  
+  return(summary_table)
+  
+}
+#this function adds a summary table to a sheet
+add_summary_table_to_sheet<- function(wb,
+                                      sheet, 
+                                      summary_table,
+                                      table_start_row, 
+                                      table_start_col){
+  
+
+  
+  #add summary table to sheet
+  writeData(wb, sheet, summary_table, startRow = table_start_row, startCol = table_start_col)
+  
+  # style table - header 
+  addStyle(
+    wb,
+    sheet = sheet,
+    summaryTableTopBottomStyle,
+    rows = table_start_row,
+    cols = table_start_col:(table_start_col + ncol(summary_table) - 1 )
```

Comment:
isn't `cols = table_start_col:(table_start_col + ncol(summary_table) - 1 )`  equivalent to
```suggestion
    cols = table_start_col:ncol(summary_table)
```

Reply:
No, because this function is designed to allow you to place the table anywhere on the sheet- so  it might start on column F (for example). This is relevant because we've been printing sampled and unsampled tables side by side. Open to feedback on how the output looks, as this might not be the best place